src/globals.js

```diff
@@ -44,50 +44,51 @@ export const GlobalTimer = (() => {
   return { setGlobalTimeout, clearTimer };
 })();
 
-export const ExecutionTags = (() => {
-  global.tags = [];
+export const validateTag = (key, value, shouldLogErrors = true) => {
+  key = String(key);
+  value = String(value);
+  if (key.length < 1 || key.length > MAX_TAG_KEY_LEN) {
+    shouldLogErrors &&
+      logger.warnClient(
+        `${ADD_TAG_ERROR_MSG_PREFIX}: key length should be between 1 and ${MAX_TAG_KEY_LEN}: ${key} - ${value}`
+      );
+    return false;
+  }
+
+  if (value.length < 1 || value.length > MAX_TAG_VALUE_LEN) {
+    shouldLogErrors &&
+      logger.warnClient(
+        `${ADD_TAG_ERROR_MSG_PREFIX}: value length should be between 1 and ${MAX_TAG_VALUE_LEN}: ${key} - ${value}`
+      );
+    return false;
+  }
+  if (global.tags.length >= MAX_TAGS) {
+    shouldLogErrors &&
+      logger.warnClient(
+        `${ADD_TAG_ERROR_MSG_PREFIX}: maximum number of tags is ${MAX_TAGS}: ${key} - ${value}`
+      );
+    return false;
+  }
+  return true;
+};
 
-  const validateTag = (key, value, shouldLogErrors = true) => {
-    key = String(key);
-    value = String(value);
-    if (key.length < 1 || key.length > MAX_TAG_KEY_LEN) {
-      shouldLogErrors &&
-        logger.warnClient(
-          `${ADD_TAG_ERROR_MSG_PREFIX}: key length should be between 1 and ${MAX_TAG_KEY_LEN}: ${key} - ${value}`
-        );
-      return false;
-    }
-    if (value.length < 1 || value.length > MAX_TAG_VALUE_LEN) {
-      shouldLogErrors &&
-        logger.warnClient(
-          `${ADD_TAG_ERROR_MSG_PREFIX}: value length should be between 1 and ${MAX_TAG_VALUE_LEN}: ${key} - ${value}`
-        );
-      return false;
-    }
-    if (global.tags.length >= MAX_TAGS) {
-      shouldLogErrors &&
-        logger.warnClient(
-          `${ADD_TAG_ERROR_MSG_PREFIX}: maximum number of tags is ${MAX_TAGS}: ${key} - ${value}`
-        );
-      return false;
-    }
-    return true;
-  };
+const normalizeTag = val => (val === undefined || val === null ? null : String(val));
```

Comment:
Why are we pushing tags with null? I think that any way we drop it in our back

Reply:
This was always there..